src/api/app/helpers/webui/color_helper.rb

```diff
@@ -0,0 +1,24 @@
+module Webui::ColorHelper
+  # Calculates a text color that will contrast well with the given color
+  def contrast_text(color)
+    color = color.gsub('#','').to_i(16)
+    blue_component = color % 0x100
+    color_without_blue = ((color - blue_component) / 0x100)
+    green_component = color_without_blue % 0x100
+    red_component = ((color_without_blue - green_component) / 0x100)
```

Comment:
I'm sure you could refactor this a bit more, but this a good starting point
```suggestion
    red_component = color[1..2].to_i(16)
    green_component = color[3..4].to_i(16)
    blue_component = color[5..6].to_i(16)
```

Reply:
I want to move this out of this PR and improve it there

Reply:
See https://github.com/openSUSE/open-build-service/pull/16641

Reply:
My suggestion wasn't applied there, so I don't know what I'm looking for